sound/soc/sof/mediatek/mt8195/mt8195.c

```diff
@@ -0,0 +1,360 @@
+// SPDX-License-Identifier: (GPL-2.0-only OR BSD-3-Clause)
+//
+// Copyright(c) 2021 Mediatek Inc. All rights reserved.
+//
+// Author: YC Hung <yc.hung@mediatek.com>
+//
+
+/*
+ * Hardware interface for audio DSP on mt8195
+ */
+
+#include <linux/delay.h>
+#include <linux/firmware.h>
+#include <linux/of_address.h>
+#include <linux/of_irq.h>
+#include <linux/of_platform.h>
+#include <linux/of_reserved_mem.h>
+#include <linux/module.h>
+
+#include <sound/sof.h>
+#include <sound/sof/xtensa.h>
+#include "mt8195.h"
+#include "../adsp_helper.h"
+#include "../../ops.h"
+#include "../../sof-audio.h"
+
+static struct adsp_chip_info *adsp_info;
+
+static void *get_adsp_chip_data(void)
+{
+	return (void *)adsp_info;
+}
+
+void __iomem *get_mbox_reg_base(u32 id)
+{
+	struct adsp_chip_info *adsp;
+
+	if (id >= DSP_MBOX_NUM)
+		return NULL;
+
+	adsp = get_adsp_chip_data();
+	if (!adsp)
+		return NULL;
+
+	return adsp->va_mboxreg[id];
+}
+EXPORT_SYMBOL(get_mbox_reg_base);
+
+static int platform_parse_resource(struct platform_device *pdev, void *data)
+{
+	struct resource *res;
+	struct resource resource;
+	struct device_node *mem_region;
+	struct device *dev = &pdev->dev;
+	struct adsp_chip_info *adsp = data;
+	int i, ret;
+
+	res = &resource;
```

Comment:
optional / cosmetic: you reuse `res` for different resources below. Maybe it would be better to just use `&resource` for the part, where that's what's needed to avoid confusion